javalib/src/main/scala/java/net/URI.scala

```diff
@@ -1,33 +1,1415 @@
 package java.net
 
-final class URI(scheme: String,
-                userInfo: String,
-                host: String,
-                port: Int,
-                path: String,
-                query: String,
-                fragment: String)
-    extends Serializable {
+import java.io.IOException
+import java.io.Serializable
+import java.io.UnsupportedEncodingException
+import java.util.StringTokenizer
 
-  def getFragment(): String =
-    fragment
+object URI {
+  val unreserved: String = "_-!.~\'()*" //$NON-NLS-1$
 
-  def getHost(): String =
-    host
+  val punct: String = ",;:$&+=" //$NON-NLS-1$
 
-  def getPath(): String =
-    path
+  val reserved: String = punct + "?/[]@" //$NON-NLS-1$
 
-  def getPort(): Int =
-    port
+  val someLegal: String = unreserved + punct
 
-  def getQuery(): String =
-    query
+  val queryLegal: String = unreserved + reserved + "\\\"" //$NON-NLS-1$
 
-  def getScheme(): String =
-    scheme
+  val allLegal: String = unreserved + reserved
 
-  def getUserInfo(): String =
-    userInfo
+  /**
```

Comment:
We usually don't include javadoc as it will not be visible to anyone. If you have comments about the logic in your code, we use `// comments` to describe what's going on close the the code.